src/main/scala/strawman/collection/mutable/LinkedHashMap.scala

```diff
@@ -0,0 +1,178 @@
+package strawman
+package collection
+package mutable
+
+import scala.{Unit, None, Option, Serializable, SerialVersionUID, Some, transient}
+
+/** $factoryInfo
+ *  @define Coll `LinkedHashMap`
+ *  @define coll linked hash map
+ */
+object LinkedHashMap extends MapFactory[LinkedHashMap] {
+
+  def empty[A, B] = new LinkedHashMap[A, B]
+
+  def fromIterable[K, V](it: collection.Iterable[(K, V)]) =
+    it match {
+      case lhm: LinkedHashMap[K, V] => lhm
+      case _ => Growable.fromIterable(empty[K, V], it)
+    }
+
+  def newBuilder[K, V]() = new GrowableBuilder(empty[K, V])
+
+  /** Class for the linked hash map entry, used internally.
+    *  @since 2.8
+    */
+  final class LinkedEntry[K, V](val key: K, var value: V)
+    extends HashEntry[K, LinkedEntry[K, V]]
+      with Serializable {
+    var earlier: LinkedEntry[K, V] = null
+    var later: LinkedEntry[K, V] = null
+  }
+
+}
+
+/** This class implements mutable maps using a hashtable.
+ *  The iterator and all traversal methods of this class visit elements in the order they were inserted.
+ *
+ *  @tparam K    the type of the keys contained in this hash map.
+ *  @tparam V    the type of the values assigned to keys in this hash map.
+ *
+ *  @define Coll `LinkedHashMap`
+ *  @define coll linked hash map
+ *  @define mayNotTerminateInf
+ *  @define willNotTerminateInf
+ *  @define orderDependent
+ *  @define orderDependentFold
+ */
+@SerialVersionUID(1L)
+class LinkedHashMap[K, V]
+  extends Map[K, V]
+    with MapOps[K, V, LinkedHashMap, LinkedHashMap[K, V]]
+    with StrictOptimizedIterableOps[(K, V), Iterable, LinkedHashMap[K, V]]
+    with Serializable {
+
+  type Entry = LinkedHashMap.LinkedEntry[K, V]
+
+  private[this] val table: HashTable[K, V, Entry] =
```

Comment:
One difference with the current standard collections is that they use inheritance whereas we use delegation to manage the hash table.